### ISSUE TITLE
Apparent contradiction between the line 5 of Algorithm 1 in the paper and implementation

### ISSUE THREAD

**Issue by Justin-Tan** [ucla-vision/entropy-sgd#5]
In the paper (line 5 of Algorithm 1), the dummy weights are updated as 

x' <- x' - \eta' dx' + \sqrt{\eta'}\epsilon N(0,1)

But in both the lua and pytorch implementations instead of multiplying by the square root of the inner learning rate the update looks like this:

Lua:
dx:add(-g, xc-lx):add(wd,lx):add(noise/math.sqrt(0.5*lclr), eta)

Pytorch:
dw.add_(-g, wc-w.data).add_(eps/np.sqrt(0.5*llr), eta)

Is there a contradiction here?

**Comment by pratikac** [comment]
the weights are updated using the gradient `dw` as
    ``w <- w - llr * dw``
on [line 83](https://github.com/ucla-vision/entropy-sgd/blob/master/python/optim.py#L83). the expression on [line 80](https://github.com/ucla-vision/entropy-sgd/blob/master/python/optim.py#L80) ensures that the variance of noise `eta` is independent of the step-size `llr`.

**Comment by Justin-Tan** [comment]
Cheers, I see.
